eng/versioning/version_client.txt

```diff
@@ -63,6 +63,7 @@ com.microsoft.azure:azure-keyvault-secrets-spring-boot-starter;2.3.2;2.3.3-beta.
 com.microsoft.azure:azure-servicebus-jms-spring-boot-starter;2.3.2;2.3.3-beta.1
 com.microsoft.azure:azure-spring-boot-metrics-starter;2.3.2;2.3.3-beta.1
 com.microsoft.azure:spring-data-cosmosdb;3.0.0-beta.1;3.0.0-beta.1
+com.microsoft.spring.data.gremlin:spring-data-gremlin;2.3.0;2.3.1-beta.1
```

Comment:
This is a good time to rename this library. The previous dependency was an external_dependency but with this PR all of those external dependency references are going away and the only reference is the current version. Renaming the groupId to com.azure and the artifactId to azure-spring-data-gremlin will also allow to remove the bannedDependency Includes for com.microsoft.spring.data.gremlin from sdk/spring/azure-spring-boot/pom.xml and sdk/spring/azure-spring-boot-starter-data-gremlin/pom.xml as they'll now be under the com.azure:* blanket include.

The dependency version here is one that has been released from a different repository, and all references to the external_dependency has been removed, since this is effectively a brand new library in the repository both the current and dependency version should be 2.3.1-beta.1. If, in the future, 2.3.0 is needed by something then it would be an external dependency.

Reply:
@jialindai is it okay we do the renaming now in this PR?